gettsim/visualization.py

```diff
@@ -156,76 +109,161 @@ def plot_dag(
+        df["y1"] = df[1].map(layout_df[1])
+    df["None"] = ""
+    if len(df) == 0:
+        edge_x = []
+        edge_y = []
+    else:
+        edge_x = df[["x0", "x1", "None"]].apply(tuple, axis=1).tolist()
+        edge_x = list(reduce(operator.concat, edge_x))
+        edge_y = df[["y0", "y1", "None"]].apply(tuple, axis=1).tolist()
+        edge_y = list(reduce(operator.concat, edge_y))
+    # prepare for the arrows
+    arrows = []
+    for i in range(len(df)):
+        arrow = go.layout.Annotation(
+            x=df["x1"][i],
+            y=df["y1"][i],
+            xref="x",
+            yref="y",
+            text="",
+            showarrow=True,
+            axref="x",
+            ayref="y",
+            ax=df["x0"][i],
+            ay=df["y0"][i],
+            arrowhead=2,
+            arrowsize=2,
+            startstandoff=5,
+            standoff=5,
+            arrowcolor="gray",
         )
-        plot.add_layout(
-            Arrow(
-                end=NormalHead(**{**ARROW_KWARGS_DEFAULTS, **arrow_kwargs}),
-                x_start=x_start,
-                y_start=y_start,
-                x_end=x_end,
-                y_end=y_end,
-                **{**EDGE_KWARGS_DEFAULTS, **edge_kwargs},
-            )
+        arrows.append(arrow)
+
+    # plot the nodes, edges and arrows together
+
+    fig = go.FigureWidget(
+        layout=go.Layout(
+            showlegend=True,
+            hovermode="closest",
+            clickmode="select",
+            annotations=arrows,
+            hoverlabel_font_size=10,
+            margin={"b": 20, "l": 5, "r": 5, "t": 40},
+            xaxis={"showgrid": False, "zeroline": False, "showticklabels": False},
+            yaxis={"showgrid": False, "zeroline": False, "showticklabels": False},
+        )
+    )
+    fig.add_scatter(
+        x=edge_x,
+        y=edge_y,
+        line={"width": 0.5, "color": "blue"},
+        hoverinfo="none",
+        mode="lines",
+        showlegend=False,
+    )
+    # choose the different option for plotting
+
+    if (len(names) > 10) or (show_labels is False):
+        mode = "markers"
+        fig.add_scatter(
+            x=combo.x,
+            y=combo.y,
+            mode=mode,
+            hoverinfo="text",
+            textposition="bottom center",
+            text=list(names),
+            showlegend=show_legend,
+            marker={
+                "showscale": False,
+                "reversescale": True,
+                "color": "red",
+                "size": 15,
+            },
         )
 
-    plot.renderers.append(graph_renderer)
-
-    tools = [BoxZoomTool(), ResetTool()]
-    tools.append(TapTool(callback=OpenURL(url="@url")))
-    if tooltips:
-        tools.append(HoverTool(tooltips=TOOLTIPS))
-
-    plot.add_tools(*tools)
+    elif (len(names) <= 10) or ((len(names) > 10) and (show_labels is True)):
+        mode = "markers+text"
+
+    elif (len(names) > 10) and (show_labels is True):
```

Comment:
This elif statement will never be hit since in case `(len(names) > 10)` is True the first statement evaluates to True.

Please check these nested if statements again carefully.